R/shortfall.R

```diff
@@ -0,0 +1,129 @@
+#' Implementation Shortfall (IS)
+#' 
+#' The Implementation Shortfall (IS, or simply Shortfall) is the difference between the return of the \emph{paper portfolio} (where all units are assumed to have transacted at the manager’s ideal prices)
+#' and the return of the \emph{actual portfolio}, which contemplates actual transaction prices of executed units net of fees incurred in these transactions.
+#' 
+#' There exist several methods proposed in the literature to calculate the Shortfall.
+#' Each of them firstly considers the difference between the units quantity intended to be traded and the units that were traded, that is the quantity of units that was not traded;
+#' then it further decomposes the above-mentioned portfolios in order to introduce new metrics that provide further post-trade analysis of the trades.
+#' 
+#' The \emph{Complete execution} is the most generic case in which the IS can be calculated, here it is assumed that all the units in the paper portfolio have been executed.
+#' This is the default behaviour of the present function, when no \code{method} argument is provided. See details below.
+#' 
+#' The \emph{Perold's method} allows to express the IS as the sum of an \emph{execution cost} (before-trade), which is the difference between actual and paper prices of the transactions,
+#' and an \emph{opportunity cost}, i.e. the cost of not transacting part of the paper portfolio. 
+#' 
+#' The \emph{Wagner's method} (or "Expanded Implementation Shortfall") allows to take account of the implementation phase of trading decisions,
+#' by separating the period of interest into an \emph{investment period} (time from investment decision to beginning of trading) 
+#' and a \emph{trading period} (time from beginning of trading to end of trading). A \emph{price change} in between is obtained 
+#' and it becomes in turn an input to Perold's method, allowing us to reformulate the IS is terms of the two periods above and thus to identify 
+#' a \emph{delay related cost} and a \emph{trading related cost}, in addition to (a remainder of) Perold's opportunity cost and the transactions fees.
+#' A particular istance of Wagner's method is the so called \emph{Market Activity Implementation Shortfall}, which takes place when the trades paper prices are not provided (assumed to be unknown) and thus the corresponding arrival prices are used instead.
+#' Furthermore, Wagner's method allows for an additional decomposition of the "Delay related cost" into the \emph{opportunity delay cost} component and the \emph{trading delay cost} component.
+#' 
+#' 
+#' @param Portfolio A portfolio name that points to a portfolio object structured with initPortf()
+#' @param Symbol A string identifying the symbol to calculate the Shortfall for
+#' @param PaQty The total number of units (such as shares or contracts) intended to trade
+#' @param PaPriceStart Decision price at the beginning of trading
+#' @param PaPriceEnd Decision future price at the end of trading
```

Comment:
@jaymon0703, P_n is made available as `PaPriceEnd` (which now I notice should be better explained in the context of the documentation I wrote above), but has no default values. You suggested defaults to the closing price for the day or the price of the last executed transaction. While I haven't a solution for the former yet, how can we go for the latter? `last(txns$Txn.price)`?

Reply:
Forgive me, the suitable default will be 'last(txns$Txn.Price)' with the ideal value being the mid-price at the end of the trading horizon. All prices, during market hours, are meant to be at the mid-price. If final transaction is in a close auction, then the transaction price will be at the close price for the day anyway. I have updated that section of the issue.

Reply:
Apart from `PaPriceEnd`, for which we do have a default value, at the moment `PaPriceStart` hasn't one. In the issue you proposed to use the arrival price, but now `ArrPrice` is a param. 
Should we similarly go with `first(txns$Txn.Price)` and then add their 'default relation'?